widget/entry.go

```diff
@@ -305,6 +305,12 @@ func (e *Entry) SetReadOnly(ro bool) {
 	}
 }
 
+// IsReadOnly is for internal use, only.
```

Comment:
If we have to add public APIs that are already deprecated or not-to-be-used just to have tests be in a different package then we have done something wrong.

Reply:
Yes, we did something wrong in the first place when we introduced a dependency from an internal core package to the `widget` package.

Reply:
This is when the interface should have been introduced.